app/javascript/packs/intl-tel-input.js

```diff
@@ -6,8 +6,8 @@ const intlCode = document.querySelector('#new_phone_form_international_code');
 
 // initialise plugin
 intlTelInput(telInput, {
-  preferredCountries: ['us', 'ca'],
-  excludeCountries: ['io', 'ki', 'nf', 'nr', 'nu', 'sh', 'sx', 'tk', 'wf'],
+  preferredCountries: ['US', 'CA'],
+  onlyCountries: JSON.parse(intlCode.getAttribute('data-countries')),
```

Comment:
@aduth this feels a little fragile to me? Would love any tips you have on ensuring this doesn't break? I think the `JSON.parse` will blow up if anything is missing so at least we might be able to detect this early?

Reply:
Seems like `try` / `catch` would be the best way to protect against parse errors. `getAttribute` can also return `null`, which notably won't throw an error if passed to `JSON.parse`, and could result in runtime errors if you were to try to reference it as an array.

Otherwise, we could try to avoid having to serialize it in the first place, and somehow initialize it from the Rails view similar to [what we do with i18n](https://github.com/18F/identity-idp/blob/89b77ef58aacb7f7c5a3f3aec36f1aaf926d5569/app/assets/javascripts/i18n-strings.en.js.erb#L3).

It might also not be too much trouble to enable TypeScript checking for this file. Probably need to do some coercion since `JSON.parse` can't otherwise know what it's going to be parsing, so not quite the level of assurance we'd want.

Reply:
Example:

```js
function parseJSONArray(json) {
    let result;
    try {
        result = JSON.parse(json);
    } catch {}

    return result || [];
}

parseJSONArray('["US"]')
// ["US"]
parseJSONArray('nonsense')
// []
parseJSONArray(null)
// []
```

Reply:
I guess I'm more worried that something might break it and we otherwise wouldn't know? Like unsure if we have good test coverage? I am not convinced that just annotating types would be enough

Reply:
Which I guess is another thought --- this doesn't necessarily have to be JSON? I could just make it a comma-separate list? JSON seems pretty standard so seems easy enough as-is

Reply:
What sort of scenario would you imagine this going wrong? Values not being initialized correctly from the server?

Maybe if we detect this parsed value to not be expected (parser error or empty), we throw an error and refuse to initialize the page, then have an RSpec integration test verify this page loads and initializes as expected (presence of tel input elements, etc)?

Reply:
Good point, I'll add a script to make sure the `data-` attribute is populated as expected

Reply:
I was also kinda curious how well `intl-tel-input` handles a case where we provide it a country code it's not expecting. Appears on a quick glance of the code it should be fine:

https://github.com/jackocnr/intl-tel-input/blob/b6d7711/src/js/intlTelInput.js#L220-L222

Because the other option would have been to consider a test which pulls down [their data](https://github.com/jackocnr/intl-tel-input/blob/b6d7711fd216a83a584d43dc7fd71787f2d861ad/src/js/data.js) and compares it to ours.

Reply:
Thinking out loud a bit: This pack isn't especially reusable, since it targets the `new_phone_form` form very specifically (based on `getElementById`s above). To your point, this feels pretty fragile to me.

Better to me would be to add this script from within the partial (possibly using the new `javascript_pack_once_tags`), then update those selectors based on what we can have assurance on the partial alone. For example, `intlCode = partial.querySelector('.international-code');`

Reply:
Added a spec in e94a01725dd40943edb464afa35d76e23d622331